api/account/account.go

```diff
@@ -56,12 +56,27 @@ type Tx interface {
 // information, and this should be accepted during the construction of the
 // chain-specific transaction builder.
 type TxBuilder interface {
-	BuildTx(from, to address.Address, value, nonce, gasLimit, gasPrice pack.U256, payload pack.Bytes) (Tx, error)
+	BuildTx(ctx context.Context, from, to address.Address, value, nonce, gasLimit, gasPrice pack.U256, payload pack.Bytes) (Tx, error)
+}
+
+// The AccountInfo interface defines functionality that must expose account
+// specific information for the underlying chain.
+type AccountInfo interface {
+	// Nonce is the current nonce of this account, which must be used to build a
+	// new transaction.
+	Nonce() pack.U256
+
+	// Balance is the native token balance of this account.
+	Balance() pack.U256
 }
```

Comment:
It makes more sense to me to have these methods defined on the `Client` interface directly with an `Account` prefix on the method name: `AccountNonce(..)` and `AccountBalance(..)`. Never ignore the naming lint warning from Go.